diktat-rules/src/test/kotlin/org/cqfn/diktat/ruleset/chapter1/IdentifierNamingWarnTest.kt

```diff
@@ -592,4 +592,23 @@ class IdentifierNamingWarnTest : LintTestBase(::IdentifierNaming) {
                 LintError(7, 5, ruleId, "${CONFUSING_IDENTIFIER_NAMING.warnText()} better name is: bt, nxt", false),
                 LintError(7, 5, ruleId, "${IDENTIFIER_LENGTH.warnText()} B", false))
     }
+
+    @Test
+    @Tag(WarningNames.GENERIC_NAME)
+    fun `check generic types`() {
+        val code =
+                """
+                    interface Test<String>
+                    interface Test1<T: String>
+                    interface Test2<T : Collection<T>>
+                    interface Test3<out T>
+                    interface Test3<in T>
```

Comment:
What about `in Tr`, will it be checked correctly?

Reply:
It'll trigger. I added tests.